src/azure/storage/imds_credential.rs

```diff
@@ -0,0 +1,156 @@
+use http::{HeaderValue, Method, Request};
+use reqwest::{Client, Url};
+use serde::Deserialize;
+use std::str;
+
+const MSI_API_VERSION: &str = "2019-08-01";
+const MSI_ENDPOINT: &str = "http://169.254.169.254/metadata/identity/oauth2/token";
+
+/// Attempts authentication using a managed identity that has been assigned to the deployment environment.
+///
+/// This authentication type works in Azure VMs, App Service and Azure Functions applications, as well as the Azure Cloud Shell
+///
+/// Built up from docs at [https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol](https://docs.microsoft.com/azure/app-service/overview-managed-identity#using-the-rest-protocol)
+#[derive(Clone, Debug)]
+pub struct ImdsCredential {
```

Comment:
If we group all of our configurations into one ﻿`Config` struct, it could become too complex. In my opinion, it would be better to define different types of credentials as separate structs, which will help keep our code clean and organized. lmk if you still think a change is needed. @Xuanwo

Reply:
> In my opinion, it would be better to define different types of credentials as separate structs, which will help keep our code clean and organized.

The types of credentials are very limited and nearly never grow up. I prefer not to over design this part.